pkg/cmd/cli/describe/projectstatus.go

```diff
@@ -326,38 +339,101 @@ func indentLines(indent string, lines ...string) []string {
 	return ret
 }
 
-func describeProjectAndServer(project *projectapi.Project, server string) string {
-	if server != "" {
+type formatter interface {
+	ResourceName(obj interface{}) string
+}
+
+func namespaceNameWithType(resource, name, namespace, defaultNamespace string, noNamespace bool) string {
+	if noNamespace || namespace == defaultNamespace || len(namespace) == 0 {
+		return resource + "/" + name
+	}
+	return resource + "/" + name + "[" + namespace + "]"
```

Comment:
This is just a visualization - I'll reflect on that format but it matches
what I had been thinking.  We probably can punt on that until we actually
have support for it - right now putting it in between makes it harder to
copy it for the cli using -n

On Dec 18, 2015, at 3:00 PM, David Eads <notifications@github.com> wrote:

In pkg/cmd/cli/describe/projectstatus.go
<https://github.com/openshift/origin/pull/6389#discussion_r48063672>:

> @@ -326,38 +339,101 @@ func indentLines(indent string, lines ...string) []string {
>  	return ret
>  }
>
> -func describeProjectAndServer(project *projectapi.Project, server string) string {
> -	if server != "" {
> +type formatter interface {
> +	ResourceName(obj interface{}) string
> +}
> +
> +func namespaceNameWithType(resource, name, namespace, defaultNamespace string, noNamespace bool) string {
> +	if noNamespace || namespace == defaultNamespace || len(namespace) == 0 {
> +		return resource + "/" + name
> +	}
> +	return resource + "/" + name + "[" + namespace + "]"

Talked with @liggitt <https://github.com/liggitt>, I think we should do
resource.group[namespace]/name. Doing it that way makes

oc get resource
oc get resource.group
oc get resource[namespace]
oc get resource.group[namespace]

oc get resource/name
oc get resource.group/name
oc get resource[namespace]/name
oc get resource.group[namespace]/name

parse nicely and when sorting, items of a given resource type are grouped
by namespace instead of by name.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/6389/files#r48063672>.